tests/pass/shims/fs.rs

```diff
@@ -18,23 +17,25 @@ mod utils;
 
 fn main() {
     test_path_conversion();
-    test_file();
-    test_file_clone();
-    test_file_create_new();
-    test_seek();
-    test_metadata();
-    test_file_set_len();
-    test_file_sync();
-    test_errors();
-    test_rename();
-    // solarish needs to support readdir/readdir64 for these tests.
-    if cfg!(not(any(target_os = "solaris", target_os = "illumos"))) {
-        test_directory();
-        test_canonicalize();
+    if cfg!(not(windows)) {
+        test_file();
+        test_file_create_new();
+        test_seek();
+        test_file_clone();
+        test_metadata();
```

Comment:
Shouldn't `test_metadata` be executed since you now support getting metadata?

`test_path_conversion` doesn't test anything interesting, I think it doesn't even call any shims.

Reply:
It does, actually! It's where I started on shims, since it failed, and apparently under the hood it just calls platform `metadata`. But I'll hoist this function too

Reply:
After trying the move, it's not really possible while keeping this PR what it is - test_metadata expects the ability to delete and write to files, as it uses the standard `prepare_with_content` utility to set up a test file.

Reply:
> It does, actually! It's where I started on shims, since it failed, and apparently under the hood it just calls platform metadata. But I'll hoist this function too

Ah, it calls `is_dir`. Seems good then, thanks.